vyper/codegen/stmt.py

```diff
@@ -117,44 +118,34 @@ def parse_Log(self):
         return events.ir_node_for_log(self.stmt, event, topic_ir, data_ir, self.context)
 
     def parse_Call(self):
-        # TODO use expr.func.type.is_internal once type annotations
-        # are consistently available.
-        is_self_function = (
-            (isinstance(self.stmt.func, vy_ast.Attribute))
-            and isinstance(self.stmt.func.value, vy_ast.Name)
-            and self.stmt.func.value.id == "self"
-        )
-
         if isinstance(self.stmt.func, vy_ast.Name):
             funcname = self.stmt.func.id
             return STMT_DISPATCH_TABLE[funcname].build_IR(self.stmt, self.context)
 
-        elif isinstance(self.stmt.func, vy_ast.Attribute) and self.stmt.func.attr in (
-            "append",
-            "pop",
-        ):
-            func_type = self.stmt.func._metadata["type"]
-            if isinstance(func_type, MemberFunctionT):
-                darray = Expr(self.stmt.func.value, self.context).ir_node
-                args = [Expr(x, self.context).ir_node for x in self.stmt.args]
-                if self.stmt.func.attr == "append":
-                    # sanity checks
-                    assert len(args) == 1
-                    arg = args[0]
-                    assert isinstance(darray.typ, DArrayT)
-                    check_assign(
-                        dummy_node_for_type(darray.typ.value_type), dummy_node_for_type(arg.typ)
-                    )
-
-                    return append_dyn_array(darray, arg)
-                else:
-                    assert len(args) == 0
-                    return pop_dyn_array(darray, return_popped_item=False)
-
-        if is_self_function:
-            return self_call.ir_for_self_call(self.stmt, self.context)
-        else:
-            return external_call.ir_for_external_call(self.stmt, self.context)
+        func_type = self.stmt.func._metadata["type"]
```

Comment:
there is magic in the AST copier which avoids copying fields which start with `"_"`. i actually want to get rid of the metadata dict and replace it with real members (i started on this by adding `_expr_info` to all exprs https://github.com/vyperlang/vyper/blob/10564dcc37756f3d3684b7a91fd8f4325a38c4d8/vyper/ast/nodes.py#L747 but they haven't been populated yet), but it's a bit of a longer-term project

Reply:
Moving the calls to a single function might be useful then, like `get_metadata(node, 'key')` or `get_type(node)`

Reply:
> Moving the calls to a single function could be useful then, like `get_metadata(node, 'key')` or `get_type(node)`, so once it's populated we can change it there

yes, i was just thinking about this! `node.get_type()` could be good as well, it is less verbose. it can also handle that pesky `KeyError` when the type has not been computed yet!